packages/kolibri-common/components/SyncSchedule/ManageSyncSchedule.vue

```diff
@@ -0,0 +1,355 @@
+<template>
+
+  <ImmersivePage
+    :appBarTitle="$tr('toolbarHeader')"
+    :route="goBack"
+  >
+    <KPageContainer>
+      <KGrid gutter="48">
+
+        <KGridItem>
+          <h1>{{ $tr('manageSyncTitle') }}</h1>
+
+        </KGridItem>
+
+        <KGridItem>
+          <p>{{ $tr('introduction') }}</p>
+        </KGridItem>
+
+
+        <KGridItem :layout8="{ span: 4 }" :layout12="{ span: 6 }" class="separate">
+          <b>{{ facility.name }}</b>
+        </KGridItem>
+        <KGridItem
+          :layout="{ alignment: 'right' }"
+          :layout8="{ span: 4 }"
+          :layout12="{ span: 6 }"
+          class="separate"
+        >
+          <KButton
+            @click="deviceModal = true"
+          >
+            {{ $tr('addDevice') }}
+          </KButton>
+        </KGridItem>
+
+
+
+      </KGrid>
+
+      <!--      creating the table-->
+      <CoreTable>
+        <template #tbody>
+          <tbody v-if="savedDevices.length > 0">
+            <tr>
+              <th>{{ $tr('deviceName') }}</th>
+              <th>{{ $tr('Schedule') }}</th>
+              <th>{{ $tr('Status') }}</th>
+              <th></th>
+            </tr>
+            <tr v-for="device in savedDevices" :key="device.id">
+              <td>
+                <span>{{ device.extra_metadata.device_name }}<br>
+                  {{ device.extra_metadata.baseurl }}
+                </span>
+              </td>
+              <td>
+                <div>
+                  {{ device.extra_metadata.sync_state }}
+                </div>
+              </td>
+
+              <td v-if="data.length > 0">
```

Comment:
Here and below, this can cause an issue because data is initialized to `null`, so you would need to do an existence check on data first: `<td v-if="data && data.length > 0">`